File: projectGofive/src/app/models/user.model.ts
import { Usery } from "./getallusermodel";

  export interface Role {
    roleId: string;
    roleName?: string;
  }
  
  export interface User {
    id: number
    firstName: string
    lastName: string
    email: string
    phone: string
    roleId: number
    username: string
    password: string
    permissions: Permission[]
  }
  
  export interface Permission {
    permissionId: number
    isReadable: boolean
    isWritable: boolean
    isDeletable: boolean
  }
  
  interface PaginatedUserResponse {
    items: Usery[];
    totalItems: number;
  }
  

File: Api/Models/UserModel.cs
using Api.Data;

public class User
{
    public int UserId { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public string? Username { get; set; }
    public string? Password { get; set; }
    public DateTime CreateDate { get; set; }
    public int RoleId { get; set; }

    // เปลี่ยน ICollection เป็น IEnumerable
    public Role? Role { get; set; }
    public IEnumerable<UserPermission>? UserPermissions { get; set; }
}
